baremetal/metal3machine_manager.go

```diff
@@ -502,6 +502,9 @@ func (m *MachineManager) Delete(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	if host == nil {
+		return errors.Errorf("host not found for machine %s", m.Machine.Name)
```

Comment:
It's also possible to output log instead of returning an error here, not sure which one is better.